Panic during metrics collection?

vossibility-collector exits to command line after attempting to sync with a repo with a large commit history (elasticsearch).

`config.toml` repo section:
```
[repositories]

    [repositories.appmetrics]
    user = "RuntimeTools"
    repo = "appmetrics"
    topic = "hooks-appmetrics"
#    start_index = 8000 # We don't expect anything relevant to move before

    [repositories.elasticsearch]
    user = "elastic"
    repo = "elasticsearch"
    topic = "hooks-elasticsearch"

```

nsqlookupd:
```
[nsqlookupd] 2015/10/19 20:58:11.661887 CLIENT(172.17.42.1:38122): pinged (last ping 14.999995807s)
[nsqlookupd] 2015/10/19 20:58:13.931483 200 GET /lookup?topic=hooks-elasticsearch (172.17.0.18:60375) 183.51µs
[nsqlookupd] 2015/10/19 20:58:26.662159 CLIENT(172.17.42.1:38122): pinged (last ping 15.000269745s)
[nsqlookupd] 2015/10/19 20:58:41.661068 CLIENT(172.17.42.1:38122): pinged (last ping 14.998911881s)
[nsqlookupd] 2015/10/19 20:58:56.660780 CLIENT(172.17.42.1:38122): pinged (last ping 14.999709406s)
[nsqlookupd] 2015/10/19 20:59:06.189902 200 GET /lookup?topic=hooks-appmetrics (172.17.0.18:60376) 180.714µs
[nsqlookupd] 2015/10/19 20:59:11.662291 CLIENT(172.17.42.1:38122): pinged (last ping 15.001518501s)
[nsqlookupd] 2015/10/19 20:59:13.931690 200 GET /lookup?topic=hooks-elasticsearch (172.17.0.18:60377) 73.39µs
[nsqlookupd] 2015/10/19 20:59:26.660257 CLIENT(172.17.42.1:38122): pinged (last ping 14.997950783s)
[nsqlookupd] 2015/10/19 20:59:41.663071 CLIENT(172.17.42.1:38122): pinged (last ping 15.00282194s)
[nsqlookupd] 2015/10/19 20:59:56.660942 CLIENT(172.17.42.1:38122): pinged (last ping 14.997872223s)
[nsqlookupd] 2015/10/19 21:00:06.188534 200 GET /lookup?topic=hooks-appmetrics (172.17.0.18:60384) 31.296µs
[nsqlookupd] 2015/10/19 21:00:11.660581 CLIENT(172.17.42.1:38122): pinged (last ping 14.999649849s)
[nsqlookupd] 2015/10/19 21:00:26.661489 CLIENT(172.17.42.1:38122): pinged (last ping 15.000898188s)
[nsqlookupd] 2015/10/19 21:00:41.661046 CLIENT(172.17.42.1:38122): pinged (last ping 14.999556965s)
[nsqlookupd] 2015/10/19 21:00:56.661434 CLIENT(172.17.42.1:38122): pinged (last ping 15.000387327s)

```



output from console:
```
time="2015-10-19T21:00:00Z" level=info msg="Starting periodic sync"
time="2015-10-19T21:00:00Z" level=info msg="retrieved 6 items for appmetrics (RuntimeTools:appmetrics) (page 1)"
2015/10/19 21:00:06 INF    1 [hooks-appmetrics/ghollector] querying nsqlookupd http://lookupd:4161/lookup?topic=hooks-appmetrics
time="2015-10-19T21:00:10Z" level=warning msg="done fetching GitHub API data"
time="2015-10-19T21:00:10Z" level=warning msg="done indexing documents in Elastic Search"
time="2015-10-19T21:00:11Z" level=info msg="retrieved 100 items for elasticsearch (elastic:elasticsearch) (page 1)"
panic: send on closed channel

goroutine 1 [running]:
main.(*syncCmd).fetchRepositoryItems(0xc2080c76e0, 0xc2080be780, 0x1, 0xa, 0x8b5100, 0x4, 0x0, 0x0)
        /src/src/cmd/vossibility-collector/synccmd.go:209 +0x4f7
main.(*syncCmd).Run(0xc2080c76e0, 0xc208029530, 0x2, 0x2)
        /src/src/cmd/vossibility-collector/synccmd.go:154 +0x1c2
main.runPeriodicSync(0xc2080cc000, 0xc20802a200)
        /src/src/cmd/vossibility-collector/run.go:117 +0x40b
main.doRunCommand(0xc20808e0f0)
        /src/src/cmd/vossibility-collector/run.go:49 +0x708
github.com/codegangsta/cli.Command.Run(0x8b6600, 0x3, 0x0, 0x0, 0x0, 0x0, 0x0, 0x935dd0, 0x20, 0x0, ...)
        /src/vendor/src/github.com/codegangsta/cli/command.go:127 +0xffe
github.com/codegangsta/cli.(*App).Run(0xc20802e300, 0xc20800a000, 0x4, 0x4, 0x0, 0x0)
        /src/vendor/src/github.com/codegangsta/cli/app.go:159 +0xd7b
main.main()
        /src/src/cmd/vossibility-collector/main.go:50 +0x4eb

goroutine 5 [syscall, 34 minutes]:
os/signal.loop()
        /usr/src/go/src/os/signal/signal_unix.go:21 +0x1f
created by os/signal.init·1
        /usr/src/go/src/os/signal/signal_unix.go:27 +0x35

goroutine 152 [select]:
github.com/bitly/go-nsq.(*Consumer).rdyLoop(0xc208096300)
        /src/vendor/src/github.com/bitly/go-nsq/consumer.go:893 +0x196
created by github.com/bitly/go-nsq.NewConsumer
        /src/vendor/src/github.com/bitly/go-nsq/consumer.go:187 +0x8e5

goroutine 153 [chan receive, 32 minutes]:
github.com/bitly/go-nsq.(*Consumer).handlerLoop(0xc208096300, 0x7fdf5ff89770, 0xc2080c1d10)
        /src/vendor/src/github.com/bitly/go-nsq/consumer.go:1095 +0xa2
created by github.com/bitly/go-nsq.(*Consumer).AddConcurrentHandlers
        /src/vendor/src/github.com/bitly/go-nsq/consumer.go:1087 +0xf5

goroutine 159 [IO wait]:
net.(*pollDesc).Wait(0xc20800c680, 0x72, 0x0, 0x0)
        /usr/src/go/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc20800c680, 0x0, 0x0)
        /usr/src/go/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).Read(0xc20800c620, 0xc2080f3000, 0x1000, 0x1000, 0x0, 0x7fdf5ff88c30, 0xc2082c14d0)
        /usr/src/go/src/net/fd_unix.go:242 +0x40f
net.(*conn).Read(0xc20804c598, 0xc2080f3000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
        /usr/src/go/src/net/net.go:121 +0xdc
bufio.(*Reader).fill(0xc2080c6900)
        /usr/src/go/src/bufio/bufio.go:97 +0x1ce
bufio.(*Reader).Read(0xc2080c6900, 0xc2082c14c8, 0x4, 0x8, 0x0, 0x0, 0x0)
        /usr/src/go/src/bufio/bufio.go:174 +0x26c
github.com/bitly/go-nsq.(*Conn).Read(0xc2080c8fc0, 0xc2082c14c8, 0x4, 0x8, 0xc2082c14c4, 0x0, 0x0)
        /src/vendor/src/github.com/bitly/go-nsq/conn.go:243 +0x11a
io.ReadAtLeast(0x7fdf5ff8af40, 0xc2080c8fc0, 0xc2082c14c8, 0x4, 0x8, 0x4, 0x0, 0x0, 0x0)
        /usr/src/go/src/io/io.go:298 +0xf1
io.ReadFull(0x7fdf5ff8af40, 0xc2080c8fc0, 0xc2082c14c8, 0x4, 0x8, 0xc2080c8618, 0x0, 0x0)
        /usr/src/go/src/io/io.go:316 +0x6d
encoding/binary.Read(0x7fdf5ff8af40, 0xc2080c8fc0, 0x7fdf5ff8af68, 0xaf0cb0, 0x73ec80, 0xc2082c14c4, 0x0, 0x0)
        /usr/src/go/src/encoding/binary/binary.go:148 +0x12d
github.com/bitly/go-nsq.ReadResponse(0x7fdf5ff8af40, 0xc2080c8fc0, 0x0, 0x0, 0x0, 0x0, 0x0)
        /src/vendor/src/github.com/bitly/go-nsq/protocol.go:54 +0x103
github.com/bitly/go-nsq.ReadUnpackedResponse(0x7fdf5ff8af40, 0xc2080c8fc0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
        /src/vendor/src/github.com/bitly/go-nsq/protocol.go:91 +0x71
github.com/bitly/go-nsq.(*Conn).readLoop(0xc2080c8fc0)
        /src/vendor/src/github.com/bitly/go-nsq/conn.go:481 +0x2da
created by github.com/bitly/go-nsq.(*Conn).Connect
        /src/vendor/src/github.com/bitly/go-nsq/conn.go:181 +0x7e4

goroutine 161 [syscall, 32 minutes, locked to thread]:
runtime.goexit()
        /usr/src/go/src/runtime/asm_amd64.s:2232 +0x1

goroutine 160 [select, 32 minutes]:
github.com/bitly/go-nsq.(*Conn).writeLoop(0xc2080c8fc0)
        /src/vendor/src/github.com/bitly/go-nsq/conn.go:547 +0x952
created by github.com/bitly/go-nsq.(*Conn).Connect
        /src/vendor/src/github.com/bitly/go-nsq/conn.go:182 +0x801

goroutine 162 [select]:
github.com/bitly/go-nsq.(*Consumer).lookupdLoop(0xc208096300)
        /src/vendor/src/github.com/bitly/go-nsq/consumer.go:388 +0x327
created by github.com/bitly/go-nsq.(*Consumer).ConnectToNSQLookupd
        /src/vendor/src/github.com/bitly/go-nsq/consumer.go:333 +0x624

goroutine 163 [select]:
github.com/bitly/go-nsq.(*Consumer).rdyLoop(0xc208096000)
        /src/vendor/src/github.com/bitly/go-nsq/consumer.go:893 +0x196
created by github.com/bitly/go-nsq.NewConsumer
        /src/vendor/src/github.com/bitly/go-nsq/consumer.go:187 +0x8e5

goroutine 164 [chan receive, 32 minutes]:
github.com/bitly/go-nsq.(*Consumer).handlerLoop(0xc208096000, 0x7fdf5ff89770, 0xc2080a1140)
        /src/vendor/src/github.com/bitly/go-nsq/consumer.go:1095 +0xa2
created by github.com/bitly/go-nsq.(*Consumer).AddConcurrentHandlers
        /src/vendor/src/github.com/bitly/go-nsq/consumer.go:1087 +0xf5

goroutine 169 [IO wait]:
net.(*pollDesc).Wait(0xc20800cbc0, 0x72, 0x0, 0x0)
        /usr/src/go/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc20800cbc0, 0x0, 0x0)
        /usr/src/go/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).Read(0xc20800cb60, 0xc208101000, 0x1000, 0x1000, 0x0, 0x7fdf5ff88c30, 0xc2082c1528)
        /usr/src/go/src/net/fd_unix.go:242 +0x40f
net.(*conn).Read(0xc20804c6c0, 0xc208101000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
        /usr/src/go/src/net/net.go:121 +0xdc
bufio.(*Reader).fill(0xc2080c71a0)
        /usr/src/go/src/bufio/bufio.go:97 +0x1ce
bufio.(*Reader).Read(0xc2080c71a0, 0xc2082c1520, 0x4, 0x8, 0x0, 0x0, 0x0)
        /usr/src/go/src/bufio/bufio.go:174 +0x26c
github.com/bitly/go-nsq.(*Conn).Read(0xc2080c9b00, 0xc2082c1520, 0x4, 0x8, 0xc2082c151c, 0x0, 0x0)
        /src/vendor/src/github.com/bitly/go-nsq/conn.go:243 +0x11a
io.ReadAtLeast(0x7fdf5ff8af40, 0xc2080c9b00, 0xc2082c1520, 0x4, 0x8, 0x4, 0x0, 0x0, 0x0)
        /usr/src/go/src/io/io.go:298 +0xf1
io.ReadFull(0x7fdf5ff8af40, 0xc2080c9b00, 0xc2082c1520, 0x4, 0x8, 0xc2080c9818, 0x0, 0x0)
        /usr/src/go/src/io/io.go:316 +0x6d
encoding/binary.Read(0x7fdf5ff8af40, 0xc2080c9b00, 0x7fdf5ff8af68, 0xaf0cb0, 0x73ec80, 0xc2082c151c, 0x0, 0x0)
        /usr/src/go/src/encoding/binary/binary.go:148 +0x12d
github.com/bitly/go-nsq.ReadResponse(0x7fdf5ff8af40, 0xc2080c9b00, 0x0, 0x0, 0x0, 0x0, 0x0)
        /src/vendor/src/github.com/bitly/go-nsq/protocol.go:54 +0x103
github.com/bitly/go-nsq.ReadUnpackedResponse(0x7fdf5ff8af40, 0xc2080c9b00, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
        /src/vendor/src/github.com/bitly/go-nsq/protocol.go:91 +0x71
github.com/bitly/go-nsq.(*Conn).readLoop(0xc2080c9b00)
        /src/vendor/src/github.com/bitly/go-nsq/conn.go:481 +0x2da
created by github.com/bitly/go-nsq.(*Conn).Connect
        /src/vendor/src/github.com/bitly/go-nsq/conn.go:181 +0x7e4

goroutine 170 [select, 32 minutes]:
github.com/bitly/go-nsq.(*Conn).writeLoop(0xc2080c9b00)
        /src/vendor/src/github.com/bitly/go-nsq/conn.go:547 +0x952
created by github.com/bitly/go-nsq.(*Conn).Connect
        /src/vendor/src/github.com/bitly/go-nsq/conn.go:182 +0x801

goroutine 171 [select]:
github.com/bitly/go-nsq.(*Consumer).lookupdLoop(0xc208096000)
        /src/vendor/src/github.com/bitly/go-nsq/consumer.go:388 +0x327
created by github.com/bitly/go-nsq.(*Consumer).ConnectToNSQLookupd
        /src/vendor/src/github.com/bitly/go-nsq/consumer.go:333 +0x624

goroutine 172 [chan receive, 32 minutes]:
main.func·002(0xc20804c650)
        /src/src/cmd/vossibility-collector/run.go:80 +0x50
created by main.monitorQueues
        /src/src/cmd/vossibility-collector/run.go:83 +0x108

goroutine 173 [chan receive, 32 minutes]:
main.func·002(0xc20804c768)
        /src/src/cmd/vossibility-collector/run.go:80 +0x50
created by main.monitorQueues
        /src/src/cmd/vossibility-collector/run.go:83 +0x108

goroutine 174 [semacquire, 32 minutes]:
sync.(*WaitGroup).Wait(0xc20807d320)
        /usr/src/go/src/sync/waitgroup.go:132 +0x169
main.func·003()
        /src/src/cmd/vossibility-collector/run.go:89 +0x30
created by main.monitorQueues
        /src/src/cmd/vossibility-collector/run.go:91 +0x1ed

goroutine 518 [runnable]:
net/http.(*persistConn).readLoop(0xc2080cce70)
        /usr/src/go/src/net/http/transport.go:928 +0x9ce
created by net/http.(*Transport).dialConn
        /usr/src/go/src/net/http/transport.go:660 +0xc9f

goroutine 519 [select]:
net/http.(*persistConn).writeLoop(0xc2080cce70)
        /usr/src/go/src/net/http/transport.go:945 +0x41d
created by net/http.(*Transport).dialConn
        /usr/src/go/src/net/http/transport.go:661 +0xcbc

goroutine 469 [select]:
github.com/bitly/go-hostpool.(*epsilonGreedyHostPool).epsilonGreedyDecay(0xc208055950)
        /src/vendor/src/github.com/bitly/go-hostpool/epsilon_greedy.go:84 +0x142
created by github.com/bitly/go-hostpool.NewEpsilonGreedy
        /src/vendor/src/github.com/bitly/go-hostpool/epsilon_greedy.go:65 +0x3a3

goroutine 484 [IO wait]:
net.(*pollDesc).Wait(0xc20800ced0, 0x72, 0x0, 0x0)
        /usr/src/go/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc20800ced0, 0x0, 0x0)
        /usr/src/go/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).Read(0xc20800ce70, 0xc2082bd000, 0x1000, 0x1000, 0x0, 0x7fdf5ff88c30, 0xc2082c1400)
        /usr/src/go/src/net/fd_unix.go:242 +0x40f
net.(*conn).Read(0xc20804c610, 0xc2082bd000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
        /usr/src/go/src/net/net.go:121 +0xdc
net/http.noteEOFReader.Read(0x7fdf5ff8ae68, 0xc20804c610, 0xc2082b0318, 0xc2082bd000, 0x1000, 0x1000, 0x7bfac0, 0x0, 0x0)
        /usr/src/go/src/net/http/transport.go:1270 +0x6e
net/http.(*noteEOFReader).Read(0xc208315720, 0xc2082bd000, 0x1000, 0x1000, 0xc208010000, 0x0, 0x0)
        <autogenerated>:125 +0xd4
bufio.(*Reader).fill(0xc208057a40)
        /usr/src/go/src/bufio/bufio.go:97 +0x1ce
bufio.(*Reader).Peek(0xc208057a40, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
        /usr/src/go/src/bufio/bufio.go:132 +0xf0
net/http.(*persistConn).readLoop(0xc2082b02c0)
        /usr/src/go/src/net/http/transport.go:842 +0xa4
created by net/http.(*Transport).dialConn
        /usr/src/go/src/net/http/transport.go:660 +0xc9f

goroutine 482 [IO wait]:
net.(*pollDesc).Wait(0xc20800cfb0, 0x72, 0x0, 0x0)
        /usr/src/go/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc20800cfb0, 0x0, 0x0)
        /usr/src/go/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).Read(0xc20800cf50, 0xc2082c3000, 0x1000, 0x1000, 0x0, 0x7fdf5ff88c30, 0xc2082c0400)
        /usr/src/go/src/net/fd_unix.go:242 +0x40f
net.(*conn).Read(0xc20804c5f0, 0xc2082c3000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
        /usr/src/go/src/net/net.go:121 +0xdc
net/http.noteEOFReader.Read(0x7fdf5ff8ae68, 0xc20804c5f0, 0xc2082b03c8, 0xc2082c3000, 0x1000, 0x1000, 0x7bfac0, 0x0, 0x0)
        /usr/src/go/src/net/http/transport.go:1270 +0x6e
net/http.(*noteEOFReader).Read(0xc208315660, 0xc2082c3000, 0x1000, 0x1000, 0xc208010000, 0x0, 0x0)
        <autogenerated>:125 +0xd4
bufio.(*Reader).fill(0xc2080577a0)
        /usr/src/go/src/bufio/bufio.go:97 +0x1ce
bufio.(*Reader).Peek(0xc2080577a0, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
        /usr/src/go/src/bufio/bufio.go:132 +0xf0
net/http.(*persistConn).readLoop(0xc2082b0370)
        /usr/src/go/src/net/http/transport.go:842 +0xa4
created by net/http.(*Transport).dialConn
        /usr/src/go/src/net/http/transport.go:660 +0xc9f

goroutine 483 [select]:
net/http.(*persistConn).writeLoop(0xc2082b0370)
        /usr/src/go/src/net/http/transport.go:945 +0x41d
created by net/http.(*Transport).dialConn
        /usr/src/go/src/net/http/transport.go:661 +0xcbc

goroutine 485 [select]:
net/http.(*persistConn).writeLoop(0xc2082b02c0)
        /usr/src/go/src/net/http/transport.go:945 +0x41d
created by net/http.(*Transport).dialConn
        /usr/src/go/src/net/http/transport.go:661 +0xcbc

```

--- a/src/cmd/vossibility-collector/github/sync.go
+++ b/src/cmd/vossibility-collector/github/sync.go
@@ -169,6 +169,12 @@ func (s *syncCmd) Run(repos []*storage.Repository) {
 		// Wait until indexing completes.
 		s.wgIndex.Wait()
 		log.Warn("done indexing documents in Elastic Search")
+
+		// we've closed the channels, but if the repo array is
+		// larger than 1, we need fresh channels for the next
+		// iteration of the for loop
+		s.toFetch = make(chan github.Issue, s.options.NumFetchProcs)
+		s.toIndex = make(chan githubIndexedItem, s.options.NumIndexProcs)
 	}
 }
 

